packages/commons/src/core/ApiClientBase.ts

```diff
@@ -1,21 +1,42 @@
 import axios, { Axios, AxiosInstance, CreateAxiosDefaults } from "axios";
-import { RequestObject, RequestFunction } from "../types/index.js";
+import {
+  RequestObject,
+  RequestFunction,
+  RequestOptions,
+} from "../types/index.js";
 import { OpenAPIOperation } from "../types/index.js";
 import Request from "./Request.js";
 
 export abstract class ApiClientBase {
-  public axios: AxiosInstance;
+  public readonly axios: AxiosInstance;
+  public readonly defaultRequestOptions: RequestOptions = {};
```

Comment:
`readonly` only makes the reference not editable, the properties of the object could still be changed. Is this intended? Otherwise [`Object.freeze()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) might be a good addition.

Reply:
It is clearly intended to change the default options! See this test: https://github.com/mittwald/api-client-js/pull/177/files#diff-865da0b6f44f859a9846051aa6f738d86c5bd2d69927825060cd3821c7008817R42